tests/lib/rules/no-timers.js

```diff
@@ -0,0 +1,96 @@
+const rule = require('../../../lib/rules/no-timers');
+const MESSAGE = rule.meta.message;
+const RuleTester = require('eslint').RuleTester;
+const ruleTester = new RuleTester();
+const parserOptions = {
+  ecmaVersion: 6,
+  sourceType: 'module'
+};
+
+ruleTester.run('no-timers', rule, {
+  valid: [
+    {
+      code: `
+        export default Ember.Component.extend({
+          actions: {
+            foo() {
+              setTimeout(() => {});
+            }
+          }
+        });`,
+      parserOptions
+    },
+    {
+      code: `
+        export default Ember.Component.extend({
+          guarded() {
+            if (environment.isBrowser()) {
+              setTimeout(() => {});
+            }
+          }
+        });`,
+      parserOptions
+    },
+    {
+      code: `
+        export default Ember.Component.extend({
+          guarded() {
+            if (environment.isBrowser()) {
+              setInterval(() => {});
```

Comment:
Once FastBoot 1.0 lands, any app can [define the sandbox variables](https://github.com/ember-fastboot/fastboot/blob/master/src/index.js#L27) it wishes to use and it can use `setInterval`. It is just that they have to responsibly use it. I am not sure if it is a good idea to add this lint rule by default if the purpose of this repo is to be general purpose.